staging/src/k8s.io/client-go/plugin/pkg/client/auth/exec/exec.go

```diff
@@ -175,6 +175,9 @@ type credentials struct {
 // UpdateTransportConfig updates the transport.Config to use credentials
 // returned by the plugin.
 func (a *Authenticator) UpdateTransportConfig(c *transport.Config) error {
+	a.mu.Lock()
+	defer a.mu.Unlock()
```

Comment:
This should be further down by `a.onRotate = d.CloseAll`. It also looks like we will leak connections in this scenario because we discard any connrotation.Dialer.CloseAll that was already set on the authenticator.

Reply:
Sorry, I miss this PR for a long time.
1. I move the lock to the position around the `a.onRotate = d.CloseAll`
2. Before replace with new connrotation.Dialer, we can close all existing connections.
PR updated, PTAL, thanks

Reply:
cc @mikedanese @liggitt